### DISCUSSION TITLE
Copilot no longer completes an entire block of function code

### DISCUSSION THREAD

**Discussion by tmlewin** [community/community#16101] [Copilot Conversations]
I noticed that copilot doesn't auto complete entire functions like before. it can only auto complete one liners. I have to keep hitting tab every line multiple times till the function is autocompleted. Its started happening 2 days ago. back in the day lets says i create a function like function palindrome(){. As soon as i open the first brace, it autocompletes the entire block of code and i hit tab once and i am done. Now i have to keep it htting tab at every line of code it suggests

**Comment by waghan2** [comment]
> ns

I have the same problem

**Comment by mars747** [comment]
I have the same problem. Please advise. It's so inconvenient compared to previous behavior.

**Comment by AAAAAXfda-zz** [comment]
Same problem here, so annoying

**Comment by jcastillopino** [comment]
#13303 Already reported in this discussion

**Comment by csllc-one** [comment]
This seems to be a regression or a continued "unfixed" bug.  I am getting the following copilot suggestion:

`<nav class="navbar navbar-default">
    <div class="container-fluid">
      <div class="navbar-header">
        <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="#">Cogi Systems</a>
      </div>
      <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
        <ul class="nav navbar`


But this is an incomplete code block.  Copilot appears to have puked in the middle of generating the block.  I am  new to Copilot, but this seems to be the same issue as in this thread.


This occurred in licensed version of Copilot plugin for WebStorm:

WebStorm 2022.3.1
Build #WS-223.8214.51, built on December 20, 2022
Licensed to REDACTED
Subscription is active until March 15, 2023.
Runtime version: 17.0.5+1-b653.23 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 11.6.5
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Metal Rendering is ON
Non-Bundled Plugins:
    com.github.copilot (1.1.38.2229)

**Comment by wirjo** [comment]
Getting the same issue. Is there a way to automatically complete the entire code block rather than line by line?

**Comment by weiweirj** [comment]
对对对，至少应该提供一个开关。供用户选择吧。而不是一刀切

**Comment by grantcurell** [comment]
Still seeing this behavior in March 2023 - have not been able to find a fix.

**Comment by DanielFredriksson** [comment]
Same problem here

**Comment by FLAK-ZOSO** [comment]
Same problem here, please do something, it's no more useful if it's just another AI-based chatbot.

**Comment by Rafael-Pereira-2002** [comment]
Still broken

**Comment by eladavron** [comment]
A temporary workaround that works *most* (not all) of the time is to add instructions in the settings to always add a comment at the end of code blocks.
This causes the cutoff to at least be in the comment instead of in the code itself, but sometimes even that isn't enough.
